
Here is a quick import and export of the Chicago crime stats data set that several of us had used earlier, so that we can easily shrink it, as well as alter it as needed.

-----

First, some sense of where we're at:


In [17]:
%pwd 

'C:\\Users\\Angie\\Documents\\ANLT_233_Dynamic\\ANLT233-Dynamic_Viz\\Lessons'


In other words, if you are also at the top level of the shared GitHub repository for this class **when you launch your iPython Notebook** (and you're using Python3), you can exactly mimic the actions I'm taking here.

-----

Below I will change directories, and show a listing of what is in the directory.  By the time you are doing the work, there will likely be more files, however the files used here will not change.


In [18]:
%cd ..
%cd Lessons

C:\Users\Angie\Documents\ANLT_233_Dynamic\ANLT233-Dynamic_Viz
C:\Users\Angie\Documents\ANLT_233_Dynamic\ANLT233-Dynamic_Viz\Lessons


In [19]:
%ls data

 Volume in drive C is Windows8_OS
 Volume Serial Number is B4C7-F861

 Directory of C:\Users\Angie\Documents\ANLT_233_Dynamic\ANLT233-Dynamic_Viz\Lessons\data

09/02/2016  09:11 PM    <DIR>          .
09/02/2016  09:11 PM    <DIR>          ..
08/27/2016  10:15 AM         2,076,928 Boundaries - Community Areas (current).geojson
08/27/2016  10:15 AM           239,976 Boundaries - Community Areas (current).topojson
08/27/2016  10:16 AM        75,580,015 Chicago_crime_dataset.csv
08/27/2016  10:17 AM       163,256,928 Chicago_crime_dataset.json
08/31/2016  12:43 PM            49,544 LAenergyusage.json
08/31/2016  08:28 PM           100,573 mini_Chicago_crime_columns.json
08/31/2016  08:28 PM           153,996 mini_Chicago_crime_records.json
08/27/2016  10:17 AM             1,641 README.md
08/31/2016  09:28 PM           138,173 statespopden.json
09/02/2016  08:40 PM           198,465 statespopden_rec.json
09/02/2016  09:13 PM           113,238 statespopden2.json
08/27/2016  10:17 AM    <DIR

In [20]:
import pandas as pd

state_df = pd.read_json('./data/statespopden2.json')

In [21]:
state_df.shape

(1224, 6)

In [22]:
state_df.head(50)

abbreviation  area (sq. mi)     density  population       state  year
0              AL          52423   91.897221     4817528     Alabama  2012
1              AL          52423   91.287603     4785570     Alabama  2010
10             AL          52423   85.906778     4503491     Alabama  2003
100            CA         163707  225.776717    36961229  California  2009
1000           TN          42146  132.160703     5570045   Tennessee  1998
1001           TN          42146  128.520927     5416643   Tennessee  1996
1002           TN          42146  126.392445     5326936   Tennessee  1995
1003           TN          42146  124.126560     5231438   Tennessee  1994
1004           TN          42146  119.815451     5049742   Tennessee  1992
1005           TN          42146  121.899682     5137584   Tennessee  1993
1006           TN          42146  117.842429     4966587   Tennessee  1991
1007           TN          42146  116.131827     4894492   Tennessee  1990
1008           TX         268601   63.502202    17056755       Texas  1990
1009           TX         268601   64.772674    17398005       Texas  1991
101            CA         163707  221.434093    36250311  California  2007
1010           TX         268601   67.615578    18161612       Texas  1993
1011           TX         268601   66.119404    17759738       Texas  1992
1012           TX         268601   69.113898    18564062       Texas  1994
1013           TX         268601   70.583322    18958751       Texas  1995
1014           TX         268601   72.003984    19340342       Texas  1996
1015           TX         268601   75.046374    20157531       Texas  1998
1016           TX         268601   73.493088    19740317       Texas  1997
1017           TX         268601   77.976251    20944499       Texas  2000
1018           TX         268601   76.538136    20558220       Texas  1999
1019           TX         268601   79.372832    21319622       Texas  2001
102            CA         163707  223.596651    36604337  California  2008
1020           TX         268601   80.752957    21690325       Texas  2002
1021           TX         268601   83.372821    22394023       Texas  2004
1022           TX         268601   82.021031    22030931       Texas  2003
1023           TX         268601   84.802823    22778123       Texas  2005
1024           TX         268601   86.967584    23359580       Texas  2006
1025           TX         268601   90.502414    24309039       Texas  2008
1026           TX         268601   88.726338    23831983       Texas  2007
1027           TX         268601   98.466473    26448193       Texas  2013
1028           TX         268601   92.336816    24801761       Texas  2009
1029           TX         268601   93.987655    25245178       Texas  2010
103            CA         163707  218.854069    35827943  California  2005
1030           TX         268601   95.460959    25640909       Texas  2011
1031           TX         268601   97.024196    26060796       Texas  2012
1032           UT          84904   33.624694     2854871        Utah  2012
1033           UT          84904   33.152549     2814784        Utah  2011
1034           UT          84904   32.677188     2774424        Utah  2010
1035           UT          84904   32.076475     2723421        Utah  2009
1036           UT          84904   34.166494     2900872        Utah  2013
1037           UT          84904   30.596273     2597746        Utah  2007
1038           UT          84904   31.365177     2663029        Utah  2008
1039           UT          84904   29.745442     2525507        Utah  2006
104            CA         163707  220.034586    36021202  California  2006
1040           UT          84904   28.947034     2457719        Utah  2005
1041           UT          84904   27.797713     2360137        Utah  2003

In [23]:
state_df.dtypes

abbreviation      object
area (sq. mi)      int64
density          float64
population         int64
state             object
year               int64
dtype: object

In [24]:
#state_df.to_json('./data/statespopden.json', orient='columns')
state_df.to_json('./data/statespopden_rec.json', orient='records')

Records:

We'll have a higher level **list**, where each element in the list will consist of an observation.  The list element is a dictionary with the column as the key and its value, again, as the value.

E.g., here's a snippet from ourdata:

[
{"abbreviation":"AL",
"area (sq. mi)":52423,
"density":91.8972206856,
"population":4817528,
"state":"Alabama",
"year":2012},

{"abbreviation":"AL",
"area (sq. mi)":52423,
"density":91.2876027698,
"population":4785570,
"state":"Alabama",
"year":2010},

.
.
.
]
